### DISCUSSION TITLE
Need help making a tiny build for neogeo aes only

### DISCUSSION THREAD

**Discussion by sabudum** [mamedev/discussions#64] [Q&A]
So, I've been trying to compile MAME with only the neogeo core, but I have been unsuccessful so far.
I've been reading the MAME documentation and compilation guide, but I'm having trouble understanding it, the docs are quite unclear and incomplete.

What I need is the smallest build possible with only the neogeo aes core.

Also, why does MAME require so much space on the HD? It's almost 500 MB, since every other emulator out there is around 50 MB?

Does making a tiny build with only the neogeo core would also require that amount of data?

I'm under Bodhi Linux 7.0 (Ubuntu 22.04)

**Comment by cuavas** [comment]
There are several examples you can follow for creating a subset build:
* Source filter targets, using `.flt` files in src/mame.  You can also specify a filter file with `SOURCEFILTER`.
* Targets with a Lua file in scripts/target/mame and a matching `.lst` file in src/mame (`tiny` and `ci`).

You can use either approach as a starting point.

Most emulators don’t have comparable scope – they typically emulate a few related systems.  MAME is big because it does a lot of things.  Some of the things that increase the size include:
* Internal ROM definitions for supported systems.
* Internal input names.
* Internal display layouts for many systems (there are a lot of slot machines that add up).
* Function name symbols and unwind tables for all the code since it’s C++.

If you’re looking at a full MAME software list, the software list files also take up a fair bit of space.  Given the number of systems MAME supports, there are a lot of software dumps documented.

Are you running `strip` on the binary after compiling?  Just stripping symbols will reduce the size of the binary considerably.
